### PR TITLE
creating profile component and code restructuring

### PR DESCRIPTION
**Pull request by gbajaj91**
Describe your specific features or fixes and provide a preview link for the feature being incorporated

MWPW-147085

Resolves: [MWPW-147085](https://jira.corp.adobe.com/browse/MWPW-147085)

Test URLs:
- Before: https://main--ecc-milo--adobecom.hlx.page/
- After: https://gbajaj-MWPW-147085--ecc-milo--adobecom.hlx.page/

### CODE REVIEW

File: blocks/form-handler/controllers/venue-info-component-controller.js [hunk 1]
```diff
@@ -16,13 +16,14 @@ export function onResume() {
 export function onSubmit(component, inputMap) {
   const venueInfoVisible = component.querySelector('#checkbox-venue-info-visible').checked;
 
-  const imageFile = component.querySelector('#venue-image').files[0];
-  const imageData = imageFile ? `File name: ${imageFile.name}, Size: ${imageFile.size} bytes` : 'No file uploaded';
+  // const imageFile = component.querySelector('#venue-image').files[0];
```
Comment (gbajaj91) [inline blocks/form-handler/controllers/venue-info-component-controller.js:19]: to fix

File: blocks/profile-component/profile-component.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    const inputLabel = createTag('div', { slot: 'img-label' , class : 'img-upload-text'});
+    const paragraphs = element.querySelectorAll(':scope > p');
+    paragraphs.forEach((p) => {
+        inputLabel.append(p);
+      });
+    dropzone.append(inputLabel);
+    element.append(dropzone);
+}
+
+function decorateTitle(element){
```
Comment (gbajaj91) [inline blocks/profile-component/profile-component.js:20]: todo

File: utils/utils.js [hunk 3]
```diff
@@ -72,4 +74,43 @@ export function addRepeater(element, title) {
   tag.append(plusIcon);
 
   element.append(tag);
+}
+
+export async function decorateTextfield(row, type = 'text') {
```
Comment (qiyundai) [inline utils/utils.js:79]: I think the extraction here needs to be a little more configurable -- currently the function presume it's "info-field". We might need the function to be more flexible when it comes to options. If you are basing off of the newest stage, maybe the decorateSWCTextField(row, extraOptions) function in the event-materials-component.js can be of help?

File: blocks/profile-component/profile-component.js (hunk 2 above)
Comment (qiyundai) [reply blocks/profile-component/profile-component.js:20]: You can leave // TODO: tags in the code. That'll help us keep track of things directly in the code base.

File: blocks/form-handler/controllers/venue-info-component-controller.js (hunk 1 above)
Comment (qiyundai) [reply blocks/form-handler/controllers/venue-info-component-controller.js:19]: // FIXME: is another highlightable tag you can use

File: components/image-dropzone/image-dropzone.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import { getLibs } from '../../scripts/utils.js';
+import { styles } from './image-dropzone.css.js';
+import { getIcon } from '../../utils/utils.js';
+const { createTag } = await import(`${getLibs()}/utils/utils.js`);
+
+
+class ImageDropzone extends HTMLElement {
```
Comment (qiyundai) [inline components/image-dropzone/image-dropzone.js:7]: Can we chat again next week on this level of extractions? I would like to understand better how you are envisioning the complete integration of a dropzone with this and maybe we should try to make more headways on some other more concrete deliverables before we commit to this work.

I also want to be mindful that the more we extract, the more we start to take away from each component's own ability to handle things differently. The idea of these components is that we can do very specific things based on what form component it is.

For example, the img dropzone for event hero and thumbnail image are immediate uploads -- upload as the drag&drop happens, while the speaker / host dropzone are uploaded as separate submit for each speaker / host. 

With the repeater, It just occurred to me that we might also still have a few pieces missing:

- The repeater needs to cleverly re-index each of the cloned elements with an ID attribute.
- The repeater function needs to receive a callback to reattach any of the lost eventListeners from the process of cloning.

**Review by qiyundai — COMMENTED** [review]
Let's please re-evaluate the extraction work here. I have a feeling that we are doing this a bit too early given how uncertain some of the designs are. I even feel that we should do the first round of extract after the initial functional testing is finished.

File: utils/utils.js [hunk 5]
```diff
@@ -104,3 +106,87 @@ export function addRepeater(element, title) {
...
+  const text = placeholderCol.textContent.trim();
+  const handle = handlize(text);
+
+  const attrTextEl = createTag('div', { class: 'attr-text' }, maxLengthCol.textContent.trim());
+  const maxCharNum = maxLengthCol?.querySelector('strong')?.textContent.trim();
+  const isRequired = attrTextEl.textContent.trim().endsWith('*');
+
+  const input = createTag('sp-textfield', {
+    ...extraOptions,
+    id: `info-field-${handle}`,
```
Comment (qiyundai) [inline utils/utils.js:136]: Not sure if you want review already but just a quick note -- I think we should remove predefined IDs from these extracted functions. Similarly I think required, quiet and size are also something that should be defined in extraOptions from higher level.

Comment (gbajaj91) [reply utils/utils.js]: good point. i have made quiet and size as default, so that it can be overridden with options param values.

**Pull request merged**